assets/src/adaptivity/scripting.ts

```diff
@@ -541,6 +542,71 @@ export const templatizeText = (
   return templatizedText; // templatizeText(templatizedText, state, innerEnv);
 };
 
+export const checkExpressionsWithWrongBrackets = (value: string) => {
+  let originalValue = value;
+  const allexpression = extractAllExpressionsFromText(originalValue);
+  const lstEvaluatedExpression: Record<string, string> = {};
+  allexpression.forEach((expression) => {
+    const actualExpression = expression;
+    let result = expression.match(/{([^{^}]+)}/g) || [];
+    result = result.filter(
+      (expression) => expression.search(/app\.|variables\.|stage\.|session\./) !== -1,
+    );
+    if (result?.length) {
+      const obj: Record<string, string> = {};
+      for (let i = 0; i < result?.length; i++) {
+        obj['obj' + i] = result[i];
+        expression = expression.replace(result[i], 'obj' + i);
+      }
+      expression = expression.replace(/{/g, '(');
+      expression = expression.replace(/}/g, ')');
+      for (let i = 0; i < result?.length; i++) {
+        obj['obj' + i] = result[i];
+        expression = expression.replace('obj' + i, obj['obj' + i]);
+      }
+    }
+    lstEvaluatedExpression[actualExpression] = expression;
+  });
+  Object.keys(lstEvaluatedExpression).forEach((key) => {
+    originalValue = originalValue.replace(key, lstEvaluatedExpression[key]);
+  });
+  return originalValue;
+};
+
+export const formatExpression = (child: any): string => {
+  let updatedExpression = '';
+  //this section is to check the expression in CAPI-configData variables
+  if (child.key && typeof child.value === 'string') {
+    const evaluatedExp = checkExpressionsWithWrongBrackets(child.value);
+    if (evaluatedExp !== child.value) {
+      child.value = evaluatedExp;
+      updatedExpression = evaluatedExp;
+    }
+  } else {
+    //this section is to check the expression in text flow which can be in text flow component / MCQ options etc
```

Comment:
this is my concern about this function, since we have several branches in here to handle different types of component data. ideally I'd like to offload any specific logic required by a component to the component itself. perhaps though we'll leave that for a future refactoring activity so we can get this feature rolled out